org.civicrm.styleguide/CRM/Styleguide/Manager.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+class CRM_Styleguide_Manager {
+
+  /**
+   * @var array
+   */
+  private $all = NULL;
+
+  /**
+   * Get the definition of specific style-guide by name.
+   *
+   * @param string $name
+   * @return array|NULL
+   *   If the style-guide exists, it is an array with properties:
+   *     - name: string
+   *     - label: string
+   */
+  public function get($name) {
+    $all = $this->getAll();
+    return @$all[$name];
+  }
+
+  /**
+   * Get a list of all style guides.
+   *
+   * @return array
+   */
+  public function getAll() {
+    if ($this->all === NULL) {
+      $extPath = CRM_Core_Resources::singleton()->getPath('org.civicrm.styleguide');
+
+      $this->all = array(
```

Comment:
The `[...]` notation is nicer, but I consider the "style guide" functionality on a track for official adoption (i.e. aiming to bundle with core). As of today, that still requires PHP 5.3 compatibility -- thus `array(...)`.